tests/build_test.sh

```diff
@@ -148,8 +148,14 @@ function test_kernel_build_x86()
     "make -j ARCH=x86_64  olddefconfig --silent"
     "make -j$PARALLEL_CORES ARCH=x86_64"
   )
-
   compare_command_sequence '' "$LINENO" 'expected_cmd' "$output"
+
+  output=$(kernel_build 'TEST_MODE' --ccache | tail -n +1 | head -2) # Remove statistics output
+  declare -a expected_cmd=(
+    "make -j ARCH=x86_64  olddefconfig --silent"
+    "make CC=\"ccache gcc -fdiagnostics-color\" -j$PARALLEL_CORES ARCH=x86_64"
+  )
+  compare_command_sequence '' "($LINENO)" "expected_cmd" "$output"
```

Comment:
Create a separate test function for that.\ and add a test for `--cpu-scale` scenario.